packages/@aws-cdk/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -124,6 +156,15 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
       }
     }
 
+    //if the requestType is delete, then delete the log group created by the logRetention
+    if (event.RequestType == 'Delete') {
```

Comment:
Might as well just make this line `if (event.RequestType == 'Delete' && logDeletionPolicy == 'destroyLog')